### PR TITLE
Update general_solution.md

### PR DESCRIPTION
**Pull request by tratneshsingh**
Fix up the markup and add additional notes.

### CODE REVIEW

**Comment by danghai** [comment]
Hi @tratneshsingh , could you squash 2 commits into 1? You can update the change commit by `git commit --amend`

**Comment by tratneshsingh** [comment]
@danghai squashed multiple commits to one.

**Comment by danghai** [comment]
@tratneshsingh , It has my commit in your PR. When you want to update the latest commit. 
Go back to master, and update the latest in your local master
```
git checkout master
git fech upstream
(If you do not have upstream: git remote add upstream <URL : upstream>)
git merge upstream/master
```
And then go to your branch. 
```
git checkout <your branch>
git rebase master
```
That is the way I do.

**Comment by tratneshsingh** [comment]
Thans @hangdai

**Comment by goswami-rahul** [comment]
I am not sure if this file is needed, cause it doesn't fit in this repo (being written in Java).

**Comment by tratneshsingh** [comment]
I have converted the code in python and also it is good to add a
general_solution.md to each sub-directories if such a pattern exists.

File: backtrack/general_solution.md [hunk 1]
```diff
@@ -1,156 +1,294 @@
...
+        helper(nums, target, idx+1, temp, csum+nums[idx], res)
+
+        # Unmark
+        temp.pop()
+```
+
+## Palindrome Partitioning
+https://leetcode.com/problems/palindrome-partitioning/
+```python
+def palindrome_partition(s):
```
Comment (goswami-rahul) [inline backtrack/general_solution.md]: I don't understand the need of these codes here again. Aren't these already implemented in the main algorithm files?

**Comment by tratneshsingh** [comment]
Here, it is mentioned just to demonstrate common code pattern that is
applied to solve multiple different problems.

**Comment by tratneshsingh** [comment]
But anyhow, if you intend to remove it, it is fine too. But after looking
at this file’s earlier version, it greatly helps in understanding other
code in the directory.

Thanks,
@tratneshsingh

**Comment by SaadBenn** [comment]
@goswami-rahul @keon Do you guys still want to keep this file? I skimmed through the file and found that some of the algorithms are already implemented. It'd be redundant to have them both here. I was thinking that we should clean up a bit of the mess and keep the repo clean.

**Comment by christianbender** [comment]
@SaadBenn @goswami-rahul @keon  
I would also suggest we remove the file. The code is also more Java.

**Comment by tratneshsingh** [comment]
We should search for an solution approach in Python or pseudo code. .
>> This is what is done as part of this pull request. All java code
converted to python code with addition of generic pattern for pseudo code.

**Comment by keon** [comment]
How about we split the non-redundant algos in a separate files.
and keep the explanation as README.md in this folder?

**Comment by danghai** [comment]
Hi @tratneshsingh, are you interested in this issue #162 ?

**Comment by keon** [comment]
Sure that was kind of my goal from the beginning but never got to that. 
My friend did some good works at https://github.com/kdn251/interviews

**Comment by danghai** [comment]
@keon Before doing that, I think we should group all tests in a folder first. You can see in this PR #239

**Comment by tratneshsingh** [comment]
I could create the test_stack.py for stack module.

Will pick others as we go along.

**Comment by goswami-rahul** [comment]
So, I guess we can close this PR now, and delete the `general_solution.md`? #228

**Comment by danghai** [comment]
@tratneshsingh I create test_stack.py in #245 . Could you do test_string.py? 
@keon I see the README begin longer following some files. I think we only put title (array, linkedlist, ...) in README and then writting all content file for each folder following (source, complexity, ...) However it requires a lot of work to do. If not, we simply list file name for each folder. What do you think?

**Comment by tratneshsingh** [comment]
Sure would do that.

**Comment by goswami-rahul** [comment]
Closing this PR.

**Pull request closed without merging**